src/dispatchers/frameDispatcher.ts

```diff
@@ -44,6 +44,9 @@ export class FrameDispatcher extends Dispatcher<Frame, channels.FrameInitializer
       loadStates: Array.from(frame._subtreeLifecycleEvents),
     });
     this._frame = frame;
+    // Ensure client knows about all frames, otherwise it will not get events for those (such as FrameDetached)
+    for (const child of frame.childFrames())
```

Comment:
I think we should instead emit frameAttached for all frames in PageDispatcher constructor.

Reply:
Can you explain? I don't quite like the idea of firing 'model' events from a dispatcher.